### DISCUSSION TITLE
Auto update Sherpa then sherpa upgrade all then sherpa check

### DISCUSSION THREAD

**Discussion by gsbba** [OneCDOnly/sherpa#293] [General]
Hi,

Sorry Im a noob but wondering if there is a way for me to run something automatically at a specific time of the day that will auto update Sherpa, then run a "Sherpa upgrade all" and then run a "Sherpa check". That way I keep everything unto date without manually SSH in and doing this every week. Thanks in advance.

**Comment by OneCDOnly** [comment]
Hi @gsbba. 

Good question, I really should post a howto for this. 

You only need to run the 2nd command. `sherpa upgrade all` will also upgrade/update sherpa. 

`sherpa check` should only be run if there's an issue with an existing package. Running it unnecessarily puts your sherpa applications out-of-order while it runs. 

If you'd like to automate this, suggest creating a cron job. The wiki for this is here: https://wiki.qnap.com/wiki/Add_items_to_crontab

The specifics for your cron job will be something like this example (this will run consecutive sherpa commands once-per-week on Sunday morning at 2am:
```
0 2 * * 0 /usr/sbin/sherpa upgrade new && /usr/sbin/sherpa check
```
Please post back if you need any further assistance, I'm always happy to help where I can. 🤓

**Reply by gsbba** [reply]
Thanks One much appreciated
